src/block_cache/block_io.c

```diff
@@ -802,39 +784,15 @@ __wt_blkcache_write(WT_SESSION_IMPL *session, WT_ITEM *buf, WT_PAGE_BLOCK_META *
      * images that are created during recovery may have the write generation number less than the
      * btree base write generation number, so don't verify it.
      */
-    if (delta_count > 0) {
-        delta = ip->mem;
-        /* TODO assert on delta->write_gen once it exists. */
-        mem_size = delta->mem_size;
-    } else {
-        dsk = ip->mem;
-        WT_ASSERT(session, dsk->write_gen != 0);
-        mem_size = dsk->mem_size;
-    }
+    dsk = ip->mem;
+    WT_ASSERT(session, dsk->write_gen != 0);
+    mem_size = dsk->mem_size;
 
     WT_STAT_CONN_DSRC_INCR(session, cache_write);
     WT_STAT_CONN_DSRC_INCRV(session, cache_bytes_write, mem_size);
     WT_STAT_SESSION_INCRV(session, bytes_write, mem_size);
     (void)__wt_atomic_add64(&S2C(session)->cache->bytes_written, mem_size);
 
-    if (delta_count > 0) {
-        if (delta->type == WT_PAGE_COL_INT || delta->type == WT_PAGE_ROW_INT) {
-            WT_STAT_CONN_INCRV(session, block_byte_write_intl, delta->mem_size);
-            WT_STAT_CONN_INCRV(session, block_byte_write_intl_disk, ip->size);
-        } else {
-            WT_STAT_CONN_INCRV(session, block_byte_write_leaf, delta->mem_size);
-            WT_STAT_CONN_INCRV(session, block_byte_write_leaf_disk, ip->size);
-        }
-    } else {
```

Comment:
Should we be incrementing the stats in the else block?

Reply:
I've deleted the first conditional because it looks like the stats calculation was wrong before and we were double-counting the page size. The page header variable should belong to either a delta or a base image. Let me know if you disagree.